Library/ViewModels/ShippingRulesViewModelTests.swift

```diff
@@ -2,31 +2,133 @@ import Foundation
+  private let deselectCellAtIndex = TestObserver<Int, Never>()
+  private let flashScrollIndicators = TestObserver<Void, Never>()
+  private let reloadDataWithShippingRulesData = TestObserver<[ShippingRuleData], Never>()
+  private let reloadDataWithShippingRulesReload = TestObserver<Bool, Never>()
+  private let scrollToCellAtIndex = TestObserver<Int, Never>()
+  private let selectCellAtIndex = TestObserver<Int, Never>()
 
   override func setUp() {
     super.setUp()
 
-    self.vm.outputs.loadValues.observe(self.loadValues.observer)
+    self.vm.outputs.deselectCellAtIndex.observe(self.deselectCellAtIndex.observer)
+    self.vm.outputs.flashScrollIndicators.observe(self.flashScrollIndicators.observer)
+
+    self.vm.outputs.reloadDataWithShippingRules.map(first).observe(
+      self.reloadDataWithShippingRulesData.observer
+    )
+    self.vm.outputs.reloadDataWithShippingRules.map(second).observe(
+      self.reloadDataWithShippingRulesReload.observer
+    )
+    self.vm.outputs.scrollToCellAtIndex.observe(self.scrollToCellAtIndex.observer)
+    self.vm.outputs.selectCellAtIndex.observe(self.selectCellAtIndex.observer)
   }
 
-  func testValues() {
+  func testShippingRuleSelection() {
     let project = Project.template
       |> Project.lens.country .~ Project.Country.ca
 
-    let shippingRule = ShippingRule.template
-      |> ShippingRule.lens.cost .~ 25
-      |> (ShippingRule.lens.location .. Location.lens.localizedName) .~ "Canada"
+    let shippingRule1 = ShippingRule.template
+      |> ShippingRule.lens.location .~ .canada
+
+    let shippingRule2 = ShippingRule.template
+      |> ShippingRule.lens.location .~ .greatBritain
+
+    let shippingRule3 = ShippingRule.template
+      |> ShippingRule.lens.location .~ .usa
+
+    let shippingRules = [shippingRule1, shippingRule2, shippingRule3]
+    let selectedShippingRule = shippingRule2
+
+    self.vm.inputs.configureWith(
+      project, shippingRules: shippingRules, selectedShippingRule: selectedShippingRule
+    )
+    self.vm.inputs.viewDidLoad()
+
+    self.reloadDataWithShippingRulesData.assertValues([
+      [
+        ShippingRuleData(
+          project: project, selectedShippingRule: selectedShippingRule, shippingRule: shippingRule1
+        ),
+        ShippingRuleData(
+          project: project, selectedShippingRule: selectedShippingRule, shippingRule: shippingRule2
+        ),
+        ShippingRuleData(
+          project: project, selectedShippingRule: selectedShippingRule, shippingRule: shippingRule3
+        )
+      ]
+    ])
+
+    self.flashScrollIndicators.assertValueCount(1)
+    self.reloadDataWithShippingRulesData.assertValueCount(1)
+    self.reloadDataWithShippingRulesReload.assertValues([true])
+    self.scrollToCellAtIndex.assertValues([1])
+
+    self.vm.inputs.didSelectShippingRule(at: 0)
+
+    self.flashScrollIndicators.assertValueCount(1)
+    self.reloadDataWithShippingRulesData.assertValueCount(2)
```

Comment:
Should we also assert the actual values of `reloadDataWithShippingRulesData`? Mostly to check that `ShippingRuleData` is updated with the correct `selectedShippingRule`.

Reply:
Yep, I can do that!